bgpd/bgp_evpn.h

```diff
@@ -211,4 +211,9 @@ extern void bgp_evpn_init(struct bgp *bgp);
 extern int bgp_evpn_get_type5_prefixlen(const struct prefix *pfx);
 extern bool bgp_evpn_is_prefix_nht_supported(const struct prefix *pfx);
 extern void update_advertise_vrf_routes(struct bgp *bgp_vrf);
+extern void bgp_evpn_show_remote_ip_hash(struct hash_bucket *bucket,
```

Comment:
Instead of maintaining a remote_ip_hash in bgp and doing two types of resolution checks (in bgp and zebra) would it make sense to -
1. Setup a NHG for the overlay-index-gw-ip and request zebra to only resolve it via EVPN host routes with an SVI-ifindex (this will also eliminate the need for bgpd to maintain the SVI hash)
2. Add the prefix route via NHG setup in #1. zebra will anyway not install the route till the NHG is resolved.

PS: Maybe I am over-simplifying things? I guess for asym-irb the lack of a host route would make things tricky (putting a dependency on the neigh). If this was already considered during design review and rejected for some reason I would be curious to hear why.

Reply:
As I have explained in the doc, this feature is supported for asymmetric IRB only at this point. Please refer "Topology requirements" in doc/user/bgp.rst

As the draft mentions, overlay index must be resolved using recursive resolution using a type-2 route. So,  if the type-2 route is not present in the BGP VNI table, this prefix should not be marked as valid in BGP as we do not want to advertise the prefix to BGP neighbors in vrf and attract the traffic from the neighbors unnecessarily. I considered doing a neighbor check in Zebra nht, but I decided to add this in BGP as the draft https://tools.ietf.org/html/draft-ietf-bess-evpn-prefix-advertisement-11 specifically mentions "If the RT-5 specifies a GW IP address as the Overlay Index, recursive resolution can only be done if the NVE has received and installed an RT-2 (MAC/IP route) specifying that IP address in the IP address field of its NLRI."

I think nhg won't take care of recursively resolving the gateway IP. Is that correct?

Reply:
Yes, I have poured through that draft. I do think NHG is the right answer as it offers the indirection that overlay-index expects. Even if the current NHG support is not sufficient it can be extended.

I understand that this PR is only for aysm just want to be sure that sym doesn't require a complete re-write. I am OK with most of the PR expect for the overlay-index resolution part (which obviously is the crux of the feature!). I think the split between bgpd and zebra needs some analysis so does sym-IRB considerations. I will suggest we have a meeting to discuss that further so we can close on this PR. What do you guys think?

Reply:
@adharkar Thanks for the meeting to clarify the design. If we cannot eliminate the need for the remote_macip hash would it make sense to make this processing conditional to the presence of overlay-index config? PS: Not saying it is absolutely necessary; just wanted to see if you had considered it.

Reply:
This would be a good solution. If we know that we are not going to do overlay index processing, we do not need this hash. Let me investigate if this is possible.